app/Http/Controllers/Api/v1/StatementMultipleAPIController.php

```diff
@@ -88,48 +84,52 @@ public function store(Request $request): JsonResponse
+        
+        return response()->json(['statements' => $out], Response::HTTP_CREATED);
 
-            if (strtolower((string)config('app.env_real')) === 'production') {
-                // Bulk insert on production, the cron will index later.
-                Statement::insert($payload['statements']);
-            } else {
-                // Not production, we index at the moment.
-                $id_before = Statement::query()->orderBy('id', 'DESC')->first()->id;
-                Statement::insert($payload['statements']);
-                $id_after = Statement::query()->orderBy('id', 'DESC')->first()->id;
-
-                $statements = Statement::query()->where('id', '>=', $id_before)->where('id', '<=', $id_after)->get();
-                $this->statement_search_service->bulkIndexStatements($statements);
-            }
+    }
+
+    /**
+     * @codeCoverageIgnore We are ccovering this elsewhere
+     * @param array $payload
+     * @return void
+     */
+    private function insertAndAddPuidsToCacheAndDatabase(array $payload)
+    {
+        if (strtolower((string)config('app.env_real')) === 'production') {
+            // Bulk insert on production, the cron will index later.
+            Statement::insert($payload['statements']);
+        } else {
+            // Not production, we index at the moment.
+            $id_before = Statement::query()->orderBy('id', 'DESC')->first()->id;
+            Statement::insert($payload['statements']);
+            $id_after = Statement::query()->orderBy('id', 'DESC')->first()->id;
+
+            $statements = Statement::query()->where('id', '>=', $id_before)->where('id', '<=', $id_after)->get();
+            $this->statement_search_service->bulkIndexStatements($statements);
+        }
 
 
-            //No error, add the platform unique ids into the cache and database
-            foreach ($payload['statements'] as $statement) {
-                try {
-                    $this->platform_unique_id_service->addPuidToCache($statement['platform_id'], $statement['puid']);
-                } catch (PuidNotUniqueSingleException $puidNotUniqueSingleException) {
-                    Log::info('PUID Not Unique in Cache Exception thrown in Multiple Statements', [
-                        'platform_id' => $statement['platform_id'],
-                        'puid' => $statement['puid']
-                    ]);
-                }
-
-                try {
-                    $this->platform_unique_id_service->addPuidToDatabase($statement['platform_id'], $statement['puid']);
-                } catch (PuidNotUniqueSingleException $puidNotUniqueSingleException) {
-                    Log::info('PUID Not Unique in Database Exception thrown in Multiple Statements', [
-                        'platform_id' => $statement['platform_id'],
-                        'puid' => $statement['puid']
-                    ]);
-                }
+        //No error, add the platform unique ids into the cache and database
```

Comment:
These also got put into the private method because getting the puid checker mocked up is out of scope.